test/steady_state.jl

```diff
@@ -427,14 +427,39 @@ end
     test_loss(p, prob4, alg = DynamicSS(Rodas5()))
     test_loss(p, prob2, alg = SimpleNewtonRaphson())
 
+    function enzyme_gradient(p, prob; alg = NewtonRaphson())
+        dp = Enzyme.make_zero(p)
+        dprob = Enzyme.make_zero(prob)
+        Enzyme.autodiff(Reverse, (p, prob) -> test_loss(p, prob, alg=alg), Active, Duplicated(p,dp), Duplicated(prob,dprob))
```

Comment:
You should just pass Alg as an extra arg if possible rather than making a closure over it